lib/message/encrypt.js

```diff
@@ -1,40 +1,76 @@
-/* eslint-disable no-prototype-builtins */
-import { openpgp } from '../openpgp';
+import { generateSessionKey, encrypt, sign, createMessage } from 'openpgp';
+import { readToEnd, pipe, transformPair } from '@openpgp/web-stream-tools';
 import { serverTime } from '../serverTime';
-import { createMessage } from './utils';
 
-export default async function encryptMessage({ armor = true, ...options }) {
-    if (typeof options.data === 'string') {
-        options.message = createMessage(openpgp.util.removeTrailingSpaces(options.data), options.filename);
-    }
+export default async function encryptMessage({
+    data, // TODO remove, unused
+    filename, // TODO remove , unused
+    returnSessionKey,
+    date = serverTime(),
+    armor = true,
+    detached = false,
+    ...options
+}) {
+    const sanitizedOptions = { ...options, date };
 
-    if (Uint8Array.prototype.isPrototypeOf(options.data)) {
-        options.message = createMessage(options.data, options.filename);
+    if (!options.format) {
+        sanitizedOptions.format = armor ? 'armored' : 'object'; // TODO change, remove `armor` param?
     }
 
-    options.date = typeof options.date === 'undefined' ? serverTime() : options.date;
-
-    options.armor = armor;
-
-    if (options.detached) {
-        // Create detached signature of message
-        const { signature } = await openpgp.sign({ ...options, armor: false, streaming: false });
+    if (!options.sessionKey) {
+        sanitizedOptions.sessionKey = await generateSessionKey({
+            encryptionKeys: sanitizedOptions.encryptionKeys,
+            date: sanitizedOptions.date,
+            encryptionUserIDs: sanitizedOptions.encryptionUserIDs,
+            config: sanitizedOptions.config
+        });
+    }
 
-        // Encrypt message without signing it
-        options.privateKeys = [];
-        options.returnSessionKey = true;
```

Comment:
Why would this default to true for detach-signing?

Reply:
Because we use the same session key for encrypting the detached signature (`encryptedSignature`)